src/app/_helpers/statusclassfromplugin.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { PluginExecution, PluginStatus } from '../_models';
+
+import { statusClassFromPlugin } from './statusclassfromplugin';
+
+function makePluginExecution(status: string): PluginExecution {
+  // tslint:disable: no-any
+  return ({
+    pluginStatus: status,
+    executionProgress: { errors: 0 },
+  } as any) as PluginExecution;
+}
+
+function addErrorsToPlugin(pe: PluginExecution, errors: number): PluginExecution {
+  pe.executionProgress.errors = errors;
+  return pe;
```

Comment:
you should either return a new PluginExecution with the errors or you should modify the existing PluginExecution in place. For test code it's fine to just modify it in place and not return anything. it will also make the tests a little shorter.

Reply:
Done